sktime/classification/deep_learning/base.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+"""
+Abstract base class for the Keras neural network classifiers.
+
+The reason for this class between BaseClassifier and deep_learning classifiers is
+because we can generalise tags, _predict and _predict_proba
+"""
+__author__ = ["James-Large", "ABostrom", "TonyBagnall"]
+__all__ = ["BaseDeepClassifier"]
+
+from abc import ABC, abstractmethod
+
+import numpy as np
+from sklearn.preprocessing import LabelEncoder, OneHotEncoder
+from sklearn.utils import check_random_state
+
+from sktime.classification.base import BaseClassifier
+
+
+class BaseDeepClassifier(BaseClassifier, ABC):
+    """Abstract base class for deep learning time series classifiers.
+
+    The base classifier provides a deep learning default method for
+    _predict and _predict_proba, and provides a new abstract method for building a
+    model.
+
+    Parameters
+    ----------
+    batch_size : int, default = 40
+        training batch size for the model
+
+    Arguments
+    ---------
+    self.model = None
+
+    """
+
+    _tags = {"X_inner_mtype": "numpy3D"}
+    # can probably handle multivariate??
+
+    def __init__(self, batch_size=40, random_state=None):
+        super(BaseDeepClassifier, self).__init__()
+
+        self.batch_size = batch_size
+        self.random_state = random_state
+        self.model_ = None
+
+    @abstractmethod
+    def build_model(self, input_shape, n_classes, **kwargs):
+        """Construct a compiled, un-trained, keras model that is ready for training.
+
+        Parameters
+        ----------
+        input_shape : tuple
+            The shape of the data fed into the input layer
+        n_classes: int
+            The number of classes, which shall become the size of the output
+            layer
+
+        Returns
+        -------
+        output : a compiled Keras Model
+        """
+        ...
```

Comment:
is sth missing here?

Reply:
I've removed the text output : it returns a keras Model object,